docs/design/endpoints.md

```diff
@@ -117,41 +296,53 @@ public open class EndpointProviderGenerator(
     public fun render()
 
     /**
-     * Resolve a standard library function referenced in a ruleset.
+     * Render a standard library function referenced in a rule set.
      */
-    public open fun resolveFunction(func: String)
+    public open fun renderFunction(func: String, args: List<FunctionArg>)
 }
 ```
 
 ### Middleware
-Endpoint resolution, whether generated or written by the caller, is performed per-request as a pre-operation middleware.
-At runtime, endpoint resolution is comprised of two principal tasks:
-1. construct a set of provider parameters, sourcing the values as specified by the model (context params)
-2. using the sourced parameters, call the provider to resolve to an endpoint, pointing the request to it
 
-Since endpoint resolution is performed on a per-operation basis, and the sourcing of provider parameters can vary
-per-operation, middleware must now be generated per-operation (this differs from the original implementation, where the
-endpoint resolver middleware was static across operations and its implementation was part of the runtime).
+Endpoint resolution, whether generated or written by the caller, is performed per-request as a pre-serialization
```

Comment:
if it's generating an instance of `ModifyRequestMiddleware` then it is post serialization